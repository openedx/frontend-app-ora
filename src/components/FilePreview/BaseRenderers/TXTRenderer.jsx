import React from 'react';
import PropTypes from 'prop-types';
import { rendererHooks } from './textHooks';

const TXTRenderer = ({ url, onError, onSuccess }) => {
  const { content } = rendererHooks({ url, onError, onSuccess });
  return (
    <pre className="txt-renderer">
      {content}
    </pre>
  );
};

TXTRenderer.defaultProps = {};

TXTRenderer.propTypes = {
  url: PropTypes.string.isRequired,
  onError: PropTypes.func.isRequired,
  onSuccess: PropTypes.func.isRequired,
};

export default TXTRenderer;
