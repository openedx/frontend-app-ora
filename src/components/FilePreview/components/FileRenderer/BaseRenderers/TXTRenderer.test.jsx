import { shallow } from '@edx/react-unit-test-utils';

import TXTRenderer from './TXTRenderer';

jest.mock('./textHooks', () => ({
  useTextRendererData: jest.fn().mockReturnValue({ content: 'content' }),
}));

describe('<TXTRenderer />', () => {
  const props = {
    url: 'url',
    onError: jest.fn().mockName('onError'),
    onSuccess: jest.fn().mockName('onSuccess'),
  };

  it('renders default', () => {
    const wrapper = shallow(<TXTRenderer {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
