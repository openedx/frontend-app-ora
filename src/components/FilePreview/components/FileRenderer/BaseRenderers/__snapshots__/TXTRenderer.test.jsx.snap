// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TXTRenderer /> renders default 1`] = `
<pre
  className="txt-renderer"
>
  content
</pre>
`;
